source/administration/services/hdbpp/hdbpp-es-interface.rst

```diff
@@ -0,0 +1,272 @@
++-----------------------------+------------------------------------------------------------------+
+| DefaultStrategy             | default context for a new attribute (Default to RUN)             |
++-----------------------------+------------------------------------------------------------------+
+| StatisticsTimeWindow        | timeslot for statistics in seconds                               |
++-----------------------------+------------------------------------------------------------------+
+| SubscribeRetryPeriod        | retry period for subscribe event, in seconds                     |
++-----------------------------+------------------------------------------------------------------+
+
+Table 3: Event Subscriber Class properties.
+
+The **LibConfiguration** property contains the following multi-line
+configuration parameters *host*, *user*, *password*, *dbname*, *libname*, *port*.
+Table shows example configuration parameters for backend:
+
++-------------------------------------------+
+| host=srv-log-srf.fcs.elettra.trieste.it   |
++-------------------------------------------+
+| user=hdbarchiver                          |
++-------------------------------------------+
+| password=myownpassword                    |
++-------------------------------------------+
+| dbname=hdbpp                              |
++-------------------------------------------+
+| libname=dependOnDatabase (see below)      |
++-------------------------------------------+
+| port=3306                                 |
++-------------------------------------------+
+
+Table 4: LibConfiguration parameters for database.
+
+.. note::
+    .. deprecated:: 2
+    
+    **This is information is no longer relevant as the EventSubscriber is directly linked towards the necessary backend.**
+
+    *libname* should be set to one of the following values:
+
+    libname=libhdb++mysql.so      if you intend to use HDB++ with the MySQL backend
+    libname=libhdbmysql.so        if you intend to use HDB++ with the MySQL Legacy backend
+    libname=libhdb++cassandra.so  if you intend to use HDB++ with the Cassandra backend
+    libname=libhdb++timescale.so  if you intend to use HDB++ with the TimescaleDB backend
+
+    The library specified in LibConfiguration->libname is loaded dynamically by the EventSubscriber device (e.g. *hdb++-es-srv*).
+    You need to have your LD_LIBRARY_PATH environment variable correctly set (including the directory
+    where the library you intend to use is located).
+
+    libhdb++mysql and libhdb++cassandra are just implementations of the classes defined in libhdb++ library.
+    The user can decide which implementation to use by specifying this LibConfiguration -> libname device property config parameter.
+
+    The device dynamically laods the configured library configured (using dlopen()) during the device initialization.
+    See :ref:`database-interface` section for more information.
+
```

Comment:
Not sure i remember correctly, but in a recent email exchange my understanding was the dynamic loading mechanism to be kept in place.

Reply:
I went a bit ahead of myself here, I changed it.